### PR TITLE
Disable some tests under Windows when running in simulation mode

### PR DESCRIPTION
**Pull request by aserdean**
Running tests on windows with simulation mode, i.e.:
```
set OE_SIMULATION=1
cd c:\openenclave & rmdir /S /Q build & mkdir build\x64-Debug && cd build\x64-Debug && cmake -G Ninja -DBUILD_ENCLAVES=ON ../.. && ninja && ctest -V -C Debug
```
Yields the following output from ctest:
```
75% tests passed, 26 tests failed out of 102

Total Test time (real) =  17.32 sec

The following tests did not run:
          5 - tests/aesm (Skipped)
         56 - tests/bigmalloc (Skipped)
         60 - tests/debug-unsigned (Skipped)
         61 - tests/debug-signed (Skipped)
         62 - tests/nodebug-signed (Skipped)
         63 - tests/nodebug-unsigned (Skipped)
         92 - tests/sealKey (Skipped)
         96 - tests/VectorException (Skipped)

The following tests FAILED:
          6 - tests/abortStatus (SEGFAULT)
          7 - tests/ecall (SEGFAULT)
          8 - tests/mbed/selftest (SEGFAULT)
         20 - tests/mbed/des (SEGFAULT)
         21 - tests/mbed/ecdh (SEGFAULT)
         22 - tests/mbed/ecdsa (SEGFAULT)
         23 - tests/mbed/ecp (SEGFAULT)
         24 - tests/mbed/entropy (SEGFAULT)
         28 - tests/mbed/gcm.aes256_en (SEGFAULT)
         33 - tests/mbed/hmac_drbg.no_reseed (SEGFAULT)
         35 - tests/mbed/hmac_drbg.pr (SEGFAULT)
         38 - tests/mbed/mpi (SEGFAULT)
         40 - tests/mbed/pkcs1_v15 (SEGFAULT)
         41 - tests/mbed/pkcs1_v21 (SEGFAULT)
         42 - tests/mbed/pkcs5 (SEGFAULT)
         43 - tests/mbed/pk (SEGFAULT)
         45 - tests/mbed/shax (SEGFAULT)
         47 - tests/mbed/rsa (SEGFAULT)
         49 - tests/mbed/x509parse (SEGFAULT)
         50 - tests/oethread (SEGFAULT)
         51 - tests/pthread (SEGFAULT)
         52 - tests/threadcxx (SEGFAULT)
         53 - tests/thread_local (SEGFAULT)
         54 - tests/thread_local_exported (SEGFAULT)
         64 - tests/file (SEGFAULT)
         91 - tests/SampleAppCRT (SEGFAULT)
Errors while running CTest
```
Until the tests are debugged, this patch introduces a new define called `WIN32_SIMULATION` to disable those tests.
It can be used as:
```
cd c:\openenclave & rmdir /S /Q build & mkdir build\x64-Debug && cd build\x64-Debug && cmake -G Ninja -DBUILD_ENCLAVES=ON -DWIN32_SIMULATION=1 ../.. && ninja && ctest -V -C Debug
```
Please note:
tests/file – seems to be flaky so we just disable it
tests/mbed – although not all tests are failing, we disable the entire group
Also s/\t/    /g

Signed-off-by: Alin Gabriel Serdean <aserdean@ovn.org>
Reported-at: https://github.com/Microsoft/openenclave/issues/1469

### CODE REVIEW

File: tests/CMakeLists.txt [hunk 1]
```diff
@@ -12,6 +12,11 @@ else()
     message("ENABLE_FULL_LIBCXX_TESTS not set - building some libcxx tests")
 endif()
 
+if(WIN32_SIMULATION)
```
Comment (anitagov) [inline tests/CMakeLists.txt]: WIN32_SIMULATION will need to specified as an option in CMakeLists.txt in the root folder with default value as OFF as follows:
option(WIN32_SIMULATION "Windows Simulation Mode" OFF)

File: tests/CMakeLists.txt [hunk 2]
```diff
@@ -12,6 +12,11 @@ else()
...
+  message("WIN32_SIMULATION set - skipping the following group of tests:\n"
+          "abortStatus, ecall, mbed, thread, thread_local, thread_local_no_tdata")
```
Comment (anitagov) [inline tests/CMakeLists.txt]: No need to specify the list of tests being skipped. How about? "WIN32_SIMULATION set and failing tests being skipped"

File: tests/CMakeLists.txt (hunk 1 above)
Comment (anitagov) [reply tests/CMakeLists.txt]: Also "Add a space after if"

File: tests/CMakeLists.txt [hunk 3]
```diff
@@ -24,21 +29,28 @@ endif()
 
 if (UNIX OR ADD_WINDOWS_ENCLAVE_TESTS OR USE_CLANGW)
     if (OE_SGX)
-        add_subdirectory(abortStatus)
+        # The following group does not work in simulation under WIN32
```
Comment (anitagov) [inline tests/CMakeLists.txt]: Consider revising to: "Following tests currently fail in Windows simulation mode"

File: tests/CMakeLists.txt (hunk 1 above)
Comment (aserdean) [reply tests/CMakeLists.txt]: Sure. I followed the coding style from the previous if.

File: tests/CMakeLists.txt (hunk 2 above)
Comment (aserdean) [reply tests/CMakeLists.txt]: Sounds good.

File: tests/CMakeLists.txt (hunk 3 above)
Comment (aserdean) [reply tests/CMakeLists.txt]: Sure.

**Comment by anitagov** [comment]
@aserdean - The description in the PR is great but will get into the git logs. We don't want to lose the vital information  -- so, how about creating a new issue with title like "Adding WIN32_SIMULATION environment variable for facilitating testing ELF enclaves on Windows" . You can reference the issue in the PR description and add a few lines that you want to get into the log.

**Comment by aserdean** [comment]
Thanks a lot for the suggestion @anitagov .
@ionutbalutoiu opened https://github.com/Microsoft/openenclave/issues/1469, thanks!

Adding @andschwa for an additional set of eyes.

**Pull request merged**